x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/action/RetryAction.java

```diff
@@ -138,7 +145,7 @@ public boolean equals(Object obj) {
                 return false;
             }
             Request other = (Request) obj;
-            return Arrays.equals(indices, other.indices);
+            return Objects.deepEquals(indices, other.indices) && Objects.equals(indicesOptions, other.indicesOptions);
```

Comment:
nit: Could you add a new line after `&&` as it makes debugging easier since you can breakpoint on a specific check in the equals method?